plasma_framework/python_tests/tests/contracts/root_chain/test_challenge_in_flight_exit_input_spent.py

```diff
@@ -20,6 +21,97 @@ def test_challenge_in_flight_exit_input_spent_should_succeed(testlang, period):
     assert not in_flight_exit.input_piggybacked(0)
 
 
+
+@pytest.mark.parametrize("num_outputs", [1, 2, 3, 4])
+def test_challenge_in_flight_exit_should_succeed_for_all_output_indexes(testlang, num_outputs):
```

Comment:
So this test actually would not fail even without the fix

Reply:
I think not that surprisingly:
 - the problem was with checking just the spending condition
 - the index of the double spend in the challenge tx is always 0, where owner[0] signs
 - this coincides with the input transaction having owner[0]'s output on position 0.

So even though the double spend would not necessarily be of owner[0] and at position 0, the spending condition would always pass.

Reply:
> this coincides with the input transaction having owner[0]'s output on position 0.

Yes, but other than 0, the index would mismatch. I was expecting this to break.

However because we are only using `outputType` from the output we retrieve, even the index mismatches, it would end up being same output type since we have only one type atm.